assets/app/scripts/controllers/attachPersistentVolumeClaim.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateRouteController
+ * @description
+ * # CreateRouteController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('AttachPersistentVolumeClaimController', function ($filter, $routeParams, $scope, $window, ApplicationGenerator, DataService, Navigate, ProjectsService) {
+    $scope.alerts = {};
+    $scope.renderOptions = {
+      hideFilterWidget: true
+    };
+    $scope.projectName = $routeParams.project;
+    $scope.deploymentConfigName = $routeParams.deploymentConfig;
+
+    $scope.attachment = {};
+
+    ProjectsService
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+
+        var orderByDisplayName = $filter('orderByDisplayName');
+
+        DataService.get("deploymentconfigs", $scope.deploymentConfigName, context).then(function(deploymentConfig) {
+          Logger.log("dc", deploymentConfig);
+          $scope.attachment.deploymentConfig = deploymentConfig;
+        });
+
+        DataService.list("persistentvolumeclaims", context, function(pvcs) {
+          $scope.pvcs = orderByDisplayName(pvcs.by("metadata.name"));
+          Logger.log("pvcs", $scope.pvcs);
+          if (!$scope.attachment.persistentVolumeClaim && $scope.pvcs.length) {
+            $scope.attachment.persistentVolumeClaim = $scope.pvcs[0];
+          }
+        });
+
```

Comment:
Totally! Moving several bits from here to a service is something I want to before removing WIP.